src/RingGrids/scaling.jl

```diff
@@ -30,4 +32,22 @@ function _scale_lat!(field::AbstractField, v::AbstractVector)
     end
 
     return field
+end
+"""
+
+function _scale_lat!(field::AbstractField, v::AbstractVector)
+    @boundscheck get_nlat(field) == length(v) || throw(DimensionMismatch(field, v))
+    
+    arch = architecture(field)
+    # Ensure worksize is always 2D by adding layer dimension dim=1
+    worksize = ndims(field) == 1 ? (size(field, 1), 1) : size(field)
+    launch!(arch, RingGridWorkOrder, worksize, scale_lat_kernel!, field, v)
+    synchronize(arch)
+    return field
+end
+
+@kernel inbounds=true function scale_lat_kernel!(field, v)
+    ij, k = @index(Global, NTuple)
+    j = field.grid.whichring[ij]
+    field[ij, k] *= convert(eltype(field), v[j])
```

Comment:
`field[ij, j] *= v[j]` should be enough, the conversion should happen implicitly given type promotion, e.g.

```julia
function f(v, a)
    v[1] *= a
end

@code_llvm f([1f0], 2)
```
will return somewhere
```julia
; ┌ @ promotion.jl:430 within `*`
; │┌ @ promotion.jl:400 within `promote`
; ││┌ @ promotion.jl:375 within `_promote`
; │││┌ @ number.jl:7 within `convert`
; ││││┌ @ float.jl:239 within `Float32`
       %3 = sitofp i64 %"a::Int64" to float
```
and I would be surprised if that's different when compiling for GPU